src/prefect/blocks/abstract.py

```diff
@@ -65,3 +67,141 @@ async def trigger(self) -> JobRun:
         Triggers a job run in an external service and returns a JobRun object
         to track the execution of the run.
         """
+
+
+class DatabaseBlock(Block, ABC):
+    """
+    The DatabaseBlock class represents a resource for interacting with a database.
+    Heavily influenced by [PEP 249](https://peps.python.org/pep-0249/).
+    Skewed towards RDBMS systems and a separate interface may be necessary
+    for non relational databases.
+
+    These blocks have the option to compose a credentials block or accept
+    credentials directly. It depends on how tightly coupled the database
+    configuration is to the credentials configuration.
+    """
```

Comment:
Some of this can go in a comment above I think. Also, added a suggested call out for implementing context management.
```suggestion
#
 TODO: This interface is heavily influenced by [PEP 249](https://peps.python.org/pep-0249/) and is skewed towards RDBMS systems. A separate interface may be necessary for non relational databases.
class DatabaseBlock(Block, ABC):
    """
    An abstract block type that represents a database and provides an interface for interacting with it.

    Blocks that implement this interface have the option to accept
    credentials directly via attributes or via a nested `CredentialsBlock`.
    Use of a nested credentials block is recommended unless credentials
    are tightly coupled to database connection configuration.
    
    Implementing either sync or async context management on `DatabaseBlock`
    implementations is recommended.
    """
```